src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int INDEX_DATE = 0;
+    private static final int INDEX_NAME = 1;
+    private static final int INDEX_HOURS_WORKED = 2;
+    private static final int INDEX_INCOME_PER_HOUR = 3;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        StringBuilder result = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        int[] employeeSalaries = new int[names.length];
+
+        for (String datum : data) {
+            String[] dataParts = datum.split(" ");
+            String date = dataParts[INDEX_DATE];
+            String name = dataParts[INDEX_NAME];
+            int hoursWorked = Integer.parseInt(dataParts[INDEX_HOURS_WORKED]);
+            int incomePerHour = Integer.parseInt(dataParts[INDEX_INCOME_PER_HOUR]);
+
+            if (isDateInRange(date, dateFrom, dateTo)) {
+                int employeeIndex = Arrays.asList(names).indexOf(name);
```

Comment:
this task could be done using 2 loops, under the hood .indexOf uses loop, and as a result, you have 3 loops, please think how to optimize solution

Reply:
Not resolved, still you have 3 loops